src/validate-params.ts

```diff
@@ -2,22 +2,22 @@ import {joinErrors, toResult, Result} from './util';
 import * as chalk from 'chalk';
 import SoyContext from './soy-context';
 import JSContext from './js-context';
+import {Config} from './config';
 
-export default function validateParams(soyContext: SoyContext, jsContext: JSContext): Result {
-  /**
-   * We just skip validation if the default class does not extend from Component.
-   * TODO: See if we can resolve and parse the parent class' STATE.
-   */
-  if (jsContext.getSuperClassImportPath() !== 'metal-component') {
-    return toResult(true);
-  }
+function getImplicitParams(className: string | null, paramsMap: any):string[] {
+  return className ? Object.keys(paramsMap)
+    .filter(regex => className.match(new RegExp(regex)))
```

Comment:
Yea, I had the same thought when writing `callToImport`, but I wanted to get it out the door quick. We should do some kind of config validation in `readConfig` I think, and throw an `Error` if there is an issue. Then we can gracefully report it to the user immediately before any validation is attempted.